graphx/src/main/scala/org/apache/spark/graphx/lib/StronglyConnectedComponents.scala

```diff
@@ -64,11 +69,20 @@ object StronglyConnectedComponents {
         // write values to sccGraph
         sccGraph = sccGraph.outerJoinVertices(finalVertices) {
           (vid, scc, opt) => opt.getOrElse(scc)
-        }
+        }.cache()
+        // take triplet to materialize vertices and edges (faster then count on both)
+        sccGraph.triplets.take(1)
```

Comment:
It's faster because it only materializes a single partition. I think you have to call something like count.